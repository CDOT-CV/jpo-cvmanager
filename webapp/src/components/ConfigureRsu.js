import { useState } from 'react'
import SnmpwalkMenu from './SnmpwalkMenu'
import SnmpsetMenu from './SnmpsetMenu'
import RsuRebootMenu from './RsuRebootMenu'
import Accordion from '@mui/material/Accordion'
import AccordionDetails from '@mui/material/AccordionDetails'
import AccordionSummary from '@mui/material/AccordionSummary'
import { useDispatch, useSelector } from 'react-redux'
import { ThemeProvider, createTheme } from '@mui/material'
import ExpandMoreIcon from '@mui/icons-material/ExpandMore'
import Typography from '@mui/material/Typography'
import { selectSelectedRsu, selectRsu } from '../slices/rsuSlice'
import { clearConfig, selectConfigList } from '../slices/configSlice'

import './css/SnmpwalkMenu.css'

const ConfigureRsu = () => {
    const dispatch = useDispatch()

    const [expanded, setExpanded] = useState(false)
    const handleChange = (panel) => (event, isExpanded) => {
        setExpanded(isExpanded ? panel : false)
    }
    const selectedRsu = useSelector(selectSelectedRsu)
    const selectedConfigList = useSelector(selectConfigList)

    return (
        <div>
            {selectedRsu && (
                <div>
                    <h2 className="snmpheader">Selected RSU Config</h2>
                    <button
                        id="toggle"
                        onClick={() => {
                            dispatch(selectRsu(''))
                        }}
                    >
                        X
                    </button>
                    <h2 className="snmpheader2">
                        Roadway: {selectedRsu.properties.primary_route}
                        <br />
                        Milepost: {String(selectedRsu.properties.milepost)}
                        <br />
                        IPv4: {selectedRsu.properties.ipv4_address}
                    </h2>
                </div>
            )}

            {selectedRsu && (
                <div id="sideBarBlock" className="accordion">
                    <ThemeProvider theme={accordionTheme}>
                        <Accordion
                            className="accordion-content"
                            expanded={
                                expanded === 'selected-rsu-current-config'
                            }
                            onChange={handleChange(
                                'selected-rsu-current-config'
                            )}
                        >
                            <AccordionSummary
                                expandIcon={
                                    <ExpandMoreIcon className="expand" />
                                }
                                aria-controls="panel1bh-content"
                                id="panel1bh-header"
                            >
                                <Typography>Current Configuration</Typography>
                            </AccordionSummary>
                            <ThemeProvider theme={innerAccordionTheme}>
                                <Accordion>
                                    <AccordionDetails>
                                        <SnmpwalkMenu />
                                    </AccordionDetails>
                                </Accordion>
                            </ThemeProvider>
                        </Accordion>
                        <Accordion
                            className="accordion-content"
                            expanded={
                                expanded === 'selected-rsu-add-msg-forwarding'
                            }
                            onChange={handleChange(
                                'selected-rsu-add-msg-forwarding'
                            )}
                        >
                            <AccordionSummary
                                expandIcon={
                                    <ExpandMoreIcon className="expand" />
                                }
                                aria-controls="panel2bh-content"
                                id="panel2bh-header"
                                className="expand"
                            >
                                <Typography>Message Forwarding</Typography>
                            </AccordionSummary>
                            <ThemeProvider theme={innerAccordionTheme}>
                                <Accordion>
                                    <AccordionDetails>
                                        <SnmpsetMenu
                                            type="single_rsu"
                                            rsuIpList={[
                                                selectedRsu.properties
                                                    .ipv4_address,
                                            ]}
                                        />
                                    </AccordionDetails>
                                </Accordion>
                            </ThemeProvider>
                        </Accordion>
                        <Accordion
                            className="accordion-content"
                            expanded={expanded === 'selected-rsu-reboot'}
                            onChange={handleChange('selected-rsu-reboot')}
                        >
                            <AccordionSummary
                                className="expand"
                                expandIcon={
                                    <ExpandMoreIcon className="expand" />
                                }
                                aria-controls="panel3bh-content"
                                id="panel3bh-header"
                            >
                                <Typography>Reboot</Typography>
                            </AccordionSummary>
                            <ThemeProvider theme={innerAccordionTheme}>
                                <Accordion>
                                    <AccordionDetails>
                                        <RsuRebootMenu />
                                    </AccordionDetails>
                                </Accordion>
                            </ThemeProvider>
                        </Accordion>
                    </ThemeProvider>
                </div>
            )}
            {selectedConfigList.length > 0 && !selectedRsu && (
                <div>
                    <div className="header-container">
                        <h2 className="snmpheader">Multiple RSU Config</h2>
                        <button
                            id="toggle"
                            onClick={() => {
                                dispatch(clearConfig())
                            }}
                        >
                            X
                        </button>
                    </div>
                    <h2 className="snmpheader2">
                        RSU IP List: {selectedConfigList.join(', ')}
                    </h2>
                </div>
            )}
            {selectedConfigList.length > 0 && !selectedRsu && (
                <div id="sideBarBlock" className="accordion">
                    <ThemeProvider theme={accordionTheme}>
                        <Accordion
                            className="accordion-content"
                            expanded={
                                expanded === 'multiple-rsu-add-msg-forwarding'
                            }
                            onChange={handleChange(
                                'multiple-rsu-add-msg-forwarding'
                            )}
                        >
                            <AccordionSummary
                                expandIcon={
                                    <ExpandMoreIcon className="expand" />
                                }
                                aria-controls="panel2bh-content"
                                id="panel2bh-header"
                                className="expand"
                            >
                                <Typography>Message Forwarding</Typography>
                            </AccordionSummary>
                            <ThemeProvider theme={innerAccordionTheme}>
                                <Accordion>
                                    <AccordionDetails>
                                        <SnmpsetMenu
                                            type="multi_rsu"
                                            rsuIpList={selectedConfigList}
                                        />
                                    </AccordionDetails>
                                </Accordion>
                            </ThemeProvider>
                        </Accordion>
                    </ThemeProvider>
                </div>
            )}
        </div>
    )
}

const accordionTheme = createTheme({
    palette: {
        text: {
            primary: '#ffffff',
            secondary: '#ffffff',
            disabled: '#ffffff',
            hint: '#ffffff',
        },
        divider: '#333',
        background: {
            paper: '#333',
        },
    },
})
const innerAccordionTheme = createTheme({
    palette: {
        text: {
            primary: '#fff',
            secondary: '#fff',
            disabled: '#fff',
            hint: '#fff',
        },
        divider: '#333',
        background: {
            paper: '#333',
        },
    },
})

export default ConfigureRsu
