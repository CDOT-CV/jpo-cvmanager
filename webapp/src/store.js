import { configureStore } from "@reduxjs/toolkit";
import rsuReducer from "./slices/rsuSlice";
import userReducer from "./slices/userSlice";
import wzdxReducer from "./slices/wzdxSlice";
import configReducer from "./slices/configSlice";

const store = configureStore({
  reducer: {
    rsu: rsuReducer,
    user: userReducer,
    wzdx: wzdxReducer,
    config: configReducer,
  },
});

export default store;
