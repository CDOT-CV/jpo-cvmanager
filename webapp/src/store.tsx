import { Action, ThunkAction, configureStore } from '@reduxjs/toolkit'
import rsuReducer from './generalSlices/rsuSlice'
import userReducer from './generalSlices/userSlice'
import wzdxReducer from './generalSlices/wzdxSlice'
import configReducer from './generalSlices/configSlice'
import adminAddOrganizationReducer from './features/adminAddOrganization/adminAddOrganizationSlice'
import adminAddRsuReducer from './features/adminAddRsu/adminAddRsuSlice'
import adminAddUserReducer from './features/adminAdduser/adminAddUserSlice'
import adminEditOrganizationReducer from './features/adminEditOrganization/adminEditOrganizationSlice'
import adminEditRsuReducer from './features/adminEditRsu/adminEditRsuSlice'
import adminEditUserReducer from './features/adminEditUser/adminEditUserSlice'
import adminOrganizationTabReducer from './features/adminOrganizationTab/adminOrganizationTabSlice'
import adminOrganizationTabUserReducer from './features/adminOrganizationTabUser/adminOrganizationTabUserSlice'
import adminOrganizationTabRsuReducer from './features/adminOrganizationTabRsu/adminOrganizationTabRsuSlice'
import adminRsuTabReducer from './features/adminRsuTab/adminRsuTabSlice'
import adminUserTabReducer from './features/adminUserTab/adminUserTabSlice'
import menuReducer from './features/menu/menuSlice'
import intersectionMapReducer from './components/map/map-slice'
import intersectionMapLayerStyleReducer from './components/map/map-layer-style-slice'

export const setupStore = (preloadedState: any) => {
  return configureStore({
    reducer: {
      rsu: rsuReducer,
      user: userReducer,
      wzdx: wzdxReducer,
      config: configReducer,
      adminAddOrganization: adminAddOrganizationReducer,
      adminAddRsu: adminAddRsuReducer,
      adminAddUser: adminAddUserReducer,
      adminEditOrganization: adminEditOrganizationReducer,
      adminEditRsu: adminEditRsuReducer,
      adminEditUser: adminEditUserReducer,
      adminOrganizationTab: adminOrganizationTabReducer,
      adminOrganizationTabUser: adminOrganizationTabUserReducer,
      adminOrganizationTabRsu: adminOrganizationTabRsuReducer,
      adminRsuTab: adminRsuTabReducer,
      adminUserTab: adminUserTabReducer,
      menu: menuReducer,
      intersectionMap: intersectionMapReducer,
      intersectionMapLayerStyle: intersectionMapLayerStyleReducer,
    },
    preloadedState,
  })
}

type AppStore = ReturnType<typeof setupStore>
export type AppState = ReturnType<AppStore['getState']>
type AppThunk<ReturnType = void> = ThunkAction<ReturnType, AppState, unknown, Action>

export type RootState = ReturnType<ReturnType<typeof setupStore>['getState']>
