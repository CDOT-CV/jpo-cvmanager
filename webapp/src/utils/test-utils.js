/**
 * This function searches for the every react-aria SSR ids in a given HTMLElement node and replace every attribute values with a static id
 *
 * This can be usefull when you're trying to generate a snapshot of components using react-aria under the hood
 *
 * @ex :
 * ```
 * const { container } = render(<Component />);
 *
 * replaceReactAriaIds(container);
 * ```
 *
 * @param container The HTMLElement node to search for SSR ids
 */
function replaceChaoticIds(container) {
  const props = [
    {
      selector: "class",
      updateFunc: (val) =>
        val.replace(
          /css-[0-9a-z]{6}-MuiTableCell-root-MuiTablePagination-root/g,
          "css-mocked-MuiTableCell-root-MuiTablePagination-root"
        ),
    },
  ];

  container.querySelectorAll("td").forEach((item) => {
    props.forEach((prop) => {
      if (item.getAttribute(prop.selector)) {
        item.setAttribute(prop.selector, prop.updateFunc(item.getAttribute(prop.selector)));
      }
    });
  });
  container.querySelectorAll("input[aria-invalid]").forEach((item) => {
    delete item["aria-invalid"];
  });
  return container;
}

export { replaceChaoticIds };
