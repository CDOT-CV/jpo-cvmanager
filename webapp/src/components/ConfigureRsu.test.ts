import React from 'react'
import { render } from '@testing-library/react'
import ConfigureRsu from './ConfigureRsu'
import { Provider } from 'react-redux'
import { setupStore } from '../store'
import { replaceChaoticIds } from '../utils/test-utils'

describe('ConfigureRsu component', () => {
  it('should take a single RSU snapshot', () => {
    const { container } = render(
      <Provider
        store={setupStore({
          rsu: {
            value: {
              selectedRsu: {
                properties: {
                  primary_route: 'primary_route',
                  milepost: 111,
                  ipv4_address: '1.1.1.1',
                },
              },
            },
          },
        })}
      >
        <ConfigureRsu />
      </Provider>
    )

    expect(replaceChaoticIds(container)).toMatchSnapshot()
  })

  it('should take a multi-RSU snapshot', () => {
    const { container } = render(
      <Provider
        store={setupStore({
          config: {
            value: {
              configList: ['1.1.1.1', '2.2.2.2', '3.3.3.3'],
            },
          },
        })}
      >
        <ConfigureRsu />
      </Provider>
    )

    expect(replaceChaoticIds(container)).toMatchSnapshot()
  })
})
