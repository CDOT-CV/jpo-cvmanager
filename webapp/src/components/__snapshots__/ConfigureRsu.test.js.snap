// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigureRsu component should take a multi-RSU snapshot 1`] = `
<div>
  <div>
    <div>
      <div
        class="header-container"
      >
        <h2
          class="snmpheader"
        >
          Multiple RSU Config
        </h2>
        <button
          id="toggle"
        >
          X
        </button>
      </div>
      <h2
        class="snmpheader2"
      >
        RSU IP List: 
        1.1.1.1, 2.2.2.2, 3.3.3.3
      </h2>
    </div>
    <div
      class="accordion"
      id="sideBarBlock"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters accordion-content css-mocked-MuiPaper-root-MuiAccordion-root"
      >
        <div
          aria-controls="panel2bh-content"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters expand css-mocked-MuiButtonBase-root-MuiAccordionSummary-root"
          id="panel2bh-header"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-mocked-MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
            >
              Message Forwarding
            </p>
          </div>
          <div
            class="MuiAccordionSummary-expandIconWrapper css-mocked-MuiAccordionSummary-expandIconWrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium expand css-mocked-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-mocked-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-mocked-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-mocked-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="panel2bh-header"
                class="MuiAccordion-region"
                id="panel2bh-content"
                role="region"
              >
                <div
                  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-mocked-MuiPaper-root-MuiAccordion-root"
                >
                  <div
                    class="MuiAccordionDetails-root css-mocked-MuiAccordionDetails-root"
                  >
                    <div
                      id="snmpdiv"
                    >
                      <h2
                        id="snmpheader"
                      >
                        Message Forwarding
                      </h2>
                      <form
                        id="snmpform"
                      >
                        <label
                          id="snmplabel"
                        >
                          <strong>
                            Destination IP:
                          </strong>
                          <input
                            id="snmpinput"
                            type="text"
                            value=""
                          />
                        </label>
                        <label
                          id="snmplabel"
                        >
                          <strong>
                            Message Type:
                          </strong>
                          <select
                            id="snmpdropdown"
                          >
                            <option
                              value="bsm"
                            >
                              BSM
                            </option>
                            <option
                              value="spat"
                            >
                              SPaT
                            </option>
                            <option
                              value="map"
                            >
                              MAP
                            </option>
                            <option
                              value="srm"
                            >
                              SRM
                            </option>
                            <option
                              value="ssm"
                            >
                              SSM
                            </option>
                          </select>
                        </label>
                      </form>
                      <button
                        id="refreshbtn"
                      >
                        Add Forwarding
                      </button>
                      <button
                        id="refreshbtn"
                      >
                        Delete Forwarding
                      </button>
                      <p
                        id="infotext"
                      />
                      <p
                        id="warningtext"
                      />
                      <div>
                        <p
                          id="snmpfiltertext"
                          margintop="40px"
                        >
                          By specifying a destination IP address along with a message type, you can add message forwarding to any of the selected RSUs or delete message forwarding from any RSUs that already have that configuration.
                        </p>
                      </div>
                      <div />
                      <div />
                    </div>
                  </div>
                  <div
                    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-mocked-MuiCollapse-root"
                    style="min-height: 0px;"
                  >
                    <div
                      class="MuiCollapse-wrapper MuiCollapse-vertical css-mocked-MuiCollapse-wrapper"
                    >
                      <div
                        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-mocked-MuiCollapse-wrapperInner"
                      >
                        <div
                          class="MuiAccordion-region"
                          role="region"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ConfigureRsu component should take a single RSU snapshot 1`] = `
<div>
  <div>
    <div>
      <h2
        class="snmpheader"
      >
        Selected RSU Config
      </h2>
      <button
        id="toggle"
      >
        X
      </button>
      <h2
        class="snmpheader2"
      >
        Roadway: 
        primary_route
        <br />
        Milepost: 
        111
        <br />
        IPv4: 
        1.1.1.1
      </h2>
    </div>
    <div
      class="accordion"
      id="sideBarBlock"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters accordion-content css-mocked-MuiPaper-root-MuiAccordion-root"
      >
        <div
          aria-controls="panel1bh-content"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-mocked-MuiButtonBase-root-MuiAccordionSummary-root"
          id="panel1bh-header"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-mocked-MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
            >
              Current Configuration
            </p>
          </div>
          <div
            class="MuiAccordionSummary-expandIconWrapper css-mocked-MuiAccordionSummary-expandIconWrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium expand css-mocked-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-mocked-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-mocked-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-mocked-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="panel1bh-header"
                class="MuiAccordion-region"
                id="panel1bh-content"
                role="region"
              >
                <div
                  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-mocked-MuiPaper-root-MuiAccordion-root"
                >
                  <div
                    class="MuiAccordionDetails-root css-mocked-MuiAccordionDetails-root"
                  >
                    <div
                      id="snmpdiv"
                    >
                      <div
                        id="msgfwddiv"
                      >
                        <h2
                          id="snmpheader"
                        >
                          Message Forwarding
                        </h2>
                        <button
                          aria-label="Refresh Message Forwarding"
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-mocked-MuiButtonBase-root-MuiIconButton-root"
                          data-mui-internal-clone-element="true"
                          tabindex="0"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mocked-MuiSvgIcon-root"
                            data-testid="RefreshIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-mocked-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <div />
                      <div />
                    </div>
                  </div>
                  <div
                    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-mocked-MuiCollapse-root"
                    style="min-height: 0px;"
                  >
                    <div
                      class="MuiCollapse-wrapper MuiCollapse-vertical css-mocked-MuiCollapse-wrapper"
                    >
                      <div
                        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-mocked-MuiCollapse-wrapperInner"
                      >
                        <div
                          class="MuiAccordion-region"
                          role="region"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters accordion-content css-mocked-MuiPaper-root-MuiAccordion-root"
      >
        <div
          aria-controls="panel2bh-content"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters expand css-mocked-MuiButtonBase-root-MuiAccordionSummary-root"
          id="panel2bh-header"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-mocked-MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
            >
              Message Forwarding
            </p>
          </div>
          <div
            class="MuiAccordionSummary-expandIconWrapper css-mocked-MuiAccordionSummary-expandIconWrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium expand css-mocked-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-mocked-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-mocked-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-mocked-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="panel2bh-header"
                class="MuiAccordion-region"
                id="panel2bh-content"
                role="region"
              >
                <div
                  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-mocked-MuiPaper-root-MuiAccordion-root"
                >
                  <div
                    class="MuiAccordionDetails-root css-mocked-MuiAccordionDetails-root"
                  >
                    <div
                      id="snmpdiv"
                    >
                      <h2
                        id="snmpheader"
                      >
                        Message Forwarding
                      </h2>
                      <form
                        id="snmpform"
                      >
                        <label
                          id="snmplabel"
                        >
                          <strong>
                            Destination IP:
                          </strong>
                          <input
                            id="snmpinput"
                            type="text"
                            value=""
                          />
                        </label>
                        <label
                          id="snmplabel"
                        >
                          <strong>
                            Message Type:
                          </strong>
                          <select
                            id="snmpdropdown"
                          >
                            <option
                              value="bsm"
                            >
                              BSM
                            </option>
                            <option
                              value="spat"
                            >
                              SPaT
                            </option>
                            <option
                              value="map"
                            >
                              MAP
                            </option>
                            <option
                              value="srm"
                            >
                              SRM
                            </option>
                            <option
                              value="ssm"
                            >
                              SSM
                            </option>
                          </select>
                        </label>
                      </form>
                      <button
                        id="refreshbtn"
                      >
                        Add Forwarding
                      </button>
                      <p
                        id="infotext"
                      />
                      <div />
                      <div />
                      <div />
                      <div />
                    </div>
                  </div>
                  <div
                    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-mocked-MuiCollapse-root"
                    style="min-height: 0px;"
                  >
                    <div
                      class="MuiCollapse-wrapper MuiCollapse-vertical css-mocked-MuiCollapse-wrapper"
                    >
                      <div
                        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-mocked-MuiCollapse-wrapperInner"
                      >
                        <div
                          class="MuiAccordion-region"
                          role="region"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters accordion-content css-mocked-MuiPaper-root-MuiAccordion-root"
      >
        <div
          aria-controls="panel3bh-content"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters expand css-mocked-MuiButtonBase-root-MuiAccordionSummary-root"
          id="panel3bh-header"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-mocked-MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
            >
              Reboot
            </p>
          </div>
          <div
            class="MuiAccordionSummary-expandIconWrapper css-mocked-MuiAccordionSummary-expandIconWrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium expand css-mocked-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-mocked-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-mocked-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-mocked-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="panel3bh-header"
                class="MuiAccordion-region"
                id="panel3bh-content"
                role="region"
              >
                <div
                  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-mocked-MuiPaper-root-MuiAccordion-root"
                >
                  <div
                    class="MuiAccordionDetails-root css-mocked-MuiAccordionDetails-root"
                  >
                    <div
                      id="snmpdiv"
                    >
                      <h2
                        id="snmpheader"
                      >
                        RSU Reboot
                      </h2>
                      <button
                        id="refreshbtn"
                      >
                        Reboot
                      </button>
                      <p
                        id="infotext"
                      >
                        Warning: This action could result in taking the RSU offline
                      </p>
                    </div>
                  </div>
                  <div
                    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-mocked-MuiCollapse-root"
                    style="min-height: 0px;"
                  >
                    <div
                      class="MuiCollapse-wrapper MuiCollapse-vertical css-mocked-MuiCollapse-wrapper"
                    >
                      <div
                        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-mocked-MuiCollapse-wrapperInner"
                      >
                        <div
                          class="MuiAccordion-region"
                          role="region"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
